src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Performance/DoNotGuardDictionaryRemoveByContainsKey.cs

```diff
@@ -0,0 +1,119 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using System.Linq;
+using Analyzer.Utilities;
+using Analyzer.Utilities.Extensions;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Operations;
+
+using Resx = Microsoft.NetCore.Analyzers.MicrosoftNetCoreAnalyzersResources;
+
+namespace Microsoft.NetCore.Analyzers.Performance
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp, LanguageNames.VisualBasic)]
+    public sealed class DoNotGuardDictionaryRemoveByContainsKey : DiagnosticAnalyzer
+    {
+        internal const string RuleId = "CA1839";
+
+        private static readonly LocalizableString s_localizableTitle = CreateResource(nameof(Resx.DoNotGuardDictionaryRemoveByContainsKeyTitle));
+        private static readonly LocalizableString s_localizableMessage = CreateResource(nameof(Resx.DoNotGuardDictionaryRemoveByContainsKeyMessage));
+        private static readonly LocalizableString s_localizableDescription = CreateResource(nameof(Resx.DoNotGuardDictionaryRemoveByContainsKeyDescription));
+
+        public const string AdditionalDocumentLocationInfoSeparator = ";;";
+
+        public const string ConditionalOperation = nameof(ConditionalOperation);
+        public const string ChildStatementOperation = nameof(ChildStatementOperation);
+
+        internal static readonly DiagnosticDescriptor Rule = DiagnosticDescriptorHelper.Create(
```

Comment:
I am wondering if it would make sense to make the call to `ContainsKey` fade out. It feels to me this rules falls into the bucket of "redundant" / "useless" calls.

Reply:
Makes sense.